intrahospital_api/loader.py

```diff
@@ -99,16 +101,54 @@ def create_rfh_patient_from_hospital_number(hospital_number, episode_category, r
     if emodels.Demographics.objects.filter(hospital_number=hospital_number).exists():
         raise ValueError('Patient with this hospital number already exists')
 
-    patient = Patient.objects.create()
+    if emodels.MergedMRN.objects.filter(mrn=hospital_number):
+        raise ValueError('MRN has already been merged into another MRN')
 
-    demographics = patient.demographics()
-    demographics.hospital_number = hospital_number
-    demographics.save()
+    rows = update_demographics.get_related_rows_for_mrn(hospital_number)
 
-    patient.create_episode(
-        category_name=episode_category.display_name,
-        start=datetime.date.today()
+    if not rows:
+        patient = Patient.objects.create()
+        patient.demographics_set.update(
+            hospital_number=hospital_number
+        )
+        patient.episode_set.create(
+            category_name=episode_category.display_name
+        )
+        load_patient(patient, run_async=run_async)
+        return patient
+
+    actives = [row for row in rows if row["ACTIVE_INACTIVE"] == emodels.MergedMRN.ACTIVE]
+    if len(actives) > 1 or len(actives) == 0:
+        # If there are multiple active rows, we can't create merges, so just create the patient
+        # If there are no active patients, we just create the patient
+        patient = Patient.objects.create()
+        patient.demographics_set.update(
+            hospital_number=hospital_number
+        )
+        patient.episode_set.create(
+            category_name=episode_category.display_name
+        )
+        load_patient(patient, run_async=run_async)
+        return patient
+
+    active_row = actives[0]
+    patient = Patient.objects.create()
+    patient.demographics_set.update(
+        hospital_number=active_row["PATIENT_NUMBER"]
     )
+    merged_mrns = []
+    for row in rows:
+        if row["ACTIVE_INACTIVE"] == emodels.MergedMRN.INACTIVE:
```

Comment:
Does this work ? I thought this would give you the active MRN not the merged one ?